x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/metric/InstrumentedSecurityActionListener.java

```diff
@@ -42,4 +42,32 @@ public static <R, C> ActionListener<AuthenticationResult<R>> wrapForAuthc(
         }), () -> metrics.recordTime(context, startTimeNano));
     }
 
+    /**
+     * A simpler variant that re-uses the Authentication Result as the context. This can be handy in situations when the attributes that are
+     * of interest are only available after the authentication is completed and not before.
+     * As a natural consequence, there will be no context available at the point of recording start time and in cases of exceptional failure
+     * @param metrics
+     * @param listener
+     * @param <R>
+     * @return
```

Comment:
nit: The javadoc for parameters are missing. In this context, it's fine if you either document or drop them.